Add adventure custom post type

DavidCramer opened the pull request:
Adds the `adventure` custom post type.

Fixes #7

DavidCramer committed:
Add adventure custom post type

DavidCramer committed:
Refactor for easy extraction later

DavidCramer commented:
Still requires the additional metafields setup to capture the extra data, but otherwise as a basis for the custom post type definition, this is ready for review.

cc @ThierryA @westonruter

westonruter commented on includes/class-amp-travel-cpt.php:
```diff
@@ -0,0 +1,129 @@
+			'show_ui'             => true,
+			'show_in_nav_menus'   => true,
+			'show_in_menu'        => true,
+			'show_in_admin_bar'   => true,
```
This is missing `show_in_rest` which is required for Gutenberg.

westonruter commented on includes/class-amp-travel-cpt.php:
```diff
@@ -0,0 +1,129 @@
+		 */
+		add_theme_support( 'post-thumbnails' );
+
+		// Image Sizes.
```
This should be a phpdoc comment to describe the filter.

westonruter reviewed (commented):
For the meta fields, we'll want to use either Gutenberg blocks that use postmeta as their attribute sources.

DavidCramer committed:
Set show_in_rest and add phpdoc to filter

DavidCramer commented:
Thanks @westonruter - I have addressed the changes you requested. Feel free to review when you have time.

DavidCramer committed:
Add plural slug for rest base.

miina commented:
@DavidCramer On the `adventure` post type -- did you also try editing in Gutenberg? When I create a new Adventure, publish it, and then change a setting (for example the content) and try to save again, I’m getting some errors + the title disappears from the view. Looks similar to behavior that got fixed [here](https://github.com/WordPress/gutenberg/pull/5738), however that was supposedly happening only if the CPT doesn't support `title`.

Wondering if that's something that only I'm experiencing or if it's an issue with Gutenberg.
![screen shot 2018-04-09 at 2 48 48 pm](https://user-images.githubusercontent.com/3294597/38496172-2c310810-3c05-11e8-8b02-b35d353387d4.png)

If you have a moment could you try if it works OK for you?

miina commented on includes/class-amp-travel-cpt.php:
```diff
@@ -0,0 +1,143 @@
+ *
+ * @package WPAMPTheme
+ */
+class AMP_Travel_CTP {
```
Looks like a typo, should probably be `_CPT` instead of `_CTP`.

DavidCramer replied:
oops yes, will fix

DavidCramer commented:
@miina I haven't tested yet in Gutenberg yet as this was just the definition of the type. But I think I should put it back to WIP and get that aspect sorted.

DavidCramer commented:
@westonruter - I initially did this just as the CPT definition. However, as @miina pointed out, it's not working yet with Gutenberg. I'll do more tests and clean it up first, then will let you know when you can review it again.

DavidCramer committed:
rename typo in class name

kienstra committed:
Merge in develop, resolve conflicts, use theme bootstrap.

Use AMP_Travel_Theme::instantiate_classes().
And change AMP_Travel_CPT::__contstruct to init().
This uses the bootstrap mechanism in the develop branch.

kienstra commented on includes/class-amp-travel-cpt.php:
```diff
@@ -0,0 +1,143 @@
+			'description'           => __( 'Adventure Custom Post Type for travel theme.', 'travel' ),
+			'public'                => true,
+			'exclude_from_search'   => true,
+			'publicly_queryable'    => true,
```
The following arguments have the default value, and are not needed:
* publicly_queryable
* show_ui
* show_in_nav_menus
* show_in_menu
* show_in_admin_bar
* capability_type
* hierarchical
* query_var
* can_export
* rest_controller_class

DavidCramer replied:
@kienstra - yup, I'll remove them.

kienstra commented on includes/class-amp-travel-cpt.php:
```diff
@@ -0,0 +1,143 @@
+			'280x158',
+			'240x135',
+			'160x90',
+			'122x67',
```
Are all of these image sizes needed? 

You know more about the specifics than I do. But it seems excessive to create 15 image sizes for every image uploaded.

DavidCramer replied:
@kienstra - no idea. but it's a good point. I took these from the `ampconf` theme as a starting point. I'll reduce it down to what I think should be needed for the current blocks, etc.

kienstra replied:
This is something that we should revisit. But I'm going to merge this PR to unblock other PRs.

kienstra reviewed (changes requested):
**Almost Ready, A Few Points**

Hi @DavidCramer,
Thanks for registering the new `'adventure'` post type. There are a few points here, one of which might not need any action.

kienstra committed:
Remove add_image_size(), default arguments for register_post_type().

We might need custom image sizes,
so continue this conversation.
But several PRs depend on this PR,
so it would help to merge it.

kienstra committed:
Merge in develop, resolve conflics, adjustements to AMP_Travel_Theme().

In that class, make all functions public.
Change __construct() to init(),
and move the requiring of class-amp-travel-cpt.php
to AMP_Travel_Theme::includes().
Thanks to David for creating that file.
This keeps all of the logic of the plugin files self-sufficient.
functions.php wil only have to require class-amp-travel-theme.php.
And call amp_travel_theme()->init().

kienstra committed:
Remove require_once of class-amp-travel-taxonomies.php.

And its instantiation.
That class isn't in this branch.

kienstra committed:
Merge in develop, resolve conflicts, require taxonomies.

Use the new includes() method to require it.
And instantiate and init() it in:
instantiate_classes().

kienstra reviewed (approved):
**Approved**

If it's alright, I applied the 2 review points I made with [this commit](https://github.com/xwp/travel/pull/33/commits/5ee6fdc32c2dac3e28a4649973bba49d8b3cfe24). 

@DavidCramer, could you please look at whether there are any more image sizes needed here? I deleted the `add_image_size()` call for now, to merge this.

DavidCramer commented:
Thanks @kienstra. Can always add additional sizes if needed later. It's all good.
